### PR TITLE
Plugin content type builder dev

### PR DESCRIPTION
**Pull request by lauriejim**

### CODE REVIEW

File: packages/strapi-plugin-content-type-builder/admin/src/components/HeaderNav/index.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      <div className={styles.headerNav}>
+        <div className={`${styles.noPaddingLeft} container-fluid `}>
+          <div className="row">
+            <div className="col-md-12">
+              <div className={styles.headerContainer}>
+                {map(this.props.links, (link, key) => {
+                  const notifActive = link.active ? <div className={styles.notifPoint} /> : '';
+                  return (
+                    <Link key={key} className={styles.headerLink} to={`${baseUrl}/${link.name}`} activeClassName={styles.linkActive}>
+                      <div></div>
```
Comment (Aurelsicoko) [inline packages/strapi-plugin-content-type-builder/admin/src/components/HeaderNav/index.js]: Is this necessary?

File: packages/strapi-plugin-content-type-builder/admin/src/components/List/index.js [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    super(props);
+    this.state = {
+      modal: false,
+      isPopUpFormValid: true,
+      requiredInputs: [],
+    };
+  }
+
+  componentDidMount() {
+    const requiredInputs = [];
```
Comment (Aurelsicoko) [inline packages/strapi-plugin-content-type-builder/admin/src/components/List/index.js]: Still not a big fan of the use of `forEach` loop to populate an array.

```js
const requiredInputs = Object.keys(this.props.sections).reduce((acc, current) => {
  acc.concat(
    this.props.sections[current].items
      .filter(item => has(item.validations, 'required'))
      .map(item => item.target)
  );
  
  return acc;
}, []);
```

File: packages/strapi-plugin-content-type-builder/admin/src/components/HeaderNav/index.js (hunk 1 above)
Comment (soupette) [reply packages/strapi-plugin-content-type-builder/admin/src/components/HeaderNav/index.js]: Yes, BTW it's a component from the settings-manager

File: packages/strapi-plugin-content-type-builder/admin/src/components/List/index.js (hunk 2 above)
Comment (soupette) [reply packages/strapi-plugin-content-type-builder/admin/src/components/List/index.js]: This code doesn't exist anymore. BTW it's a component from the settings-manager

**Pull request merged**